<?xml version="1.0"?>
<!--XML-->
<launch>
    <!-- SETUP ARGUMENTS -->
    <!-- enable/disable running rviz in launcher -->
    <arg name="rviz" default="true" />
    <arg name="camera_ns" default="i3dr_titania" />
    <arg name="cad_filepath" default="$(find i3dr_detect_3d)/data/cad/gripper_block_front.pcd" />

    <!-- Reduce point cloud size with VoxelGrid -->
    <!-- <node name="reduce_points" pkg="i3dr_detect_3d" type="reduce_points" output="screen" >
        <param name="grid_size" type="double" value="0.05" />
        <remap from="/points2" to="$(arg camera_ns)/points2" /> 
        <remap from="/points2_reduced" to="$(arg camera_ns)/points2_reduced" />
    </node> -->

    <!-- Detect cad in point cloud -->
    <node name="detect_cad" pkg="i3dr_detect_3d" type="detect_cad" output="screen" >
        <remap from="/points2" to="$(arg camera_ns)/points2" />
        <param name="cad_filepath" type="string" value="$(arg cad_filepath)" />
    </node>

    <!-- rosrun pcl_ros pointcloud_to_pcd input:=i3dr_titania/points2 _binary:=true _prefix:=/home/i3dr/catkin_ws/src/i3dr_detect_3d-ros/data/in_ -->

    <!-- DISPLAY -->
    <!-- run rviz with config file -->
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rivz" 
        args="-d $(find i3dr_detect_3d)/rviz/titania_prim.rviz"/>

</launch>